superset/common/query_context_processor.py

```diff
@@ -455,6 +455,13 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                 for metric in metric_names
             }
 
+            # When the original query has limit or offset we wont apply those
+            # to the subquery so we prevent data inconsistency due to missing records
+            # in the dataframes when performing the join
+            if query_object.row_limit or query_object.row_offset:
+                query_object_clone_dct["row_limit"] = None
```

Comment:
We should apply `row_limit` in config.py, in order to avoid a heavy query.

Reply:
makes sense, will do. Thanks